test/src/madhab_test.dart

```diff
@@ -10,7 +10,7 @@ void main() {
     expect(hanafiMadhab1 == hanafiMadhab2, isTrue);
     expect(hanafiMadhab1 == shafiMadhab1, isFalse);
 
-    Madhab madhab;
+    Madhab? madhab;
     expect(() => madhab.getShadowLength(), throwsFormatException);
```

Comment:
I think this doesn't make sense on the enum somehow.

Reply:
Only when `madhab` is not assigned, its throws format exception.  when making it non-nullable, it can throw FormatException

Reply:
We either can make this Non-nullable and remove the format exception and add one of the `Madhab` as a default value or leave it as nullable.

Reply:
@iamriajul  what's the plan for this? If we change it non-nullable, we need to assign a default value and remove the Exception.